src/main/java/com/refinedmods/refinedstorage/apiimpl/network/grid/handler/ItemGridHandler.java

```diff
@@ -89,21 +95,21 @@ public void onExtract(ServerPlayerEntity player, UUID id, int preferredSlot, int
 
         if (!took.isEmpty()) {
             if ((flags & EXTRACT_SHIFT) == EXTRACT_SHIFT) {
-                IItemHandler playerInventory = player.getCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, Direction.UP).orElse(null);
-                if (playerInventory != null) {
+                Optional<IItemHandler> playerInventory = player.getCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, Direction.UP).resolve();
```

Comment:
Why can't you do this?
```
player.getCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, Direction.UP).ifPresent(playerInventory -> {
   // your logic here
});
```

Reply:
This is what that would look like. Notices the "finalSize" and the took array. Your choice ofc. 

```java
int finalSize = size;
player.getCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, Direction.UP).ifPresent(playerInventory -> {
                    if (preferredSlot != -1) {
                        ItemStack remainder = playerInventory.insertItem(preferredSlot, took[0], true);
                        if (remainder.getCount() != took[0].getCount()) {
                            ItemStack inserted = network.extractItem(item, finalSize - remainder.getCount(), Action.PERFORM);
                            playerInventory.insertItem(preferredSlot, inserted, false);
                            took[0].setCount(remainder.getCount());
                        }
                    }
                    if (!took[0].isEmpty()) {
                        if (ItemHandlerHelper.insertItemStacked(playerInventory, took[0], true).isEmpty()) {
                            took[0] = network.extractItem(item, finalSize, Action.PERFORM);

                            ItemHandlerHelper.insertItemStacked(playerInventory, took[0], false);
                        }
                    }
                });
```

Reply:
OK, fair point. Let's leave it as it, much cleaner.